### ISSUE TITLE
Google Font Performance Boost (using font-display)

### ISSUE THREAD

**Issue by wplit** [soflyy/oxygen-bugs-and-features#337]
**Describe the feature you'd like to see included in Oxygen.**
Add option to add [font-display](https://addyosmani.com/blog/google-fonts-font-display/) to Google Fonts as an option inside global font options along side the weights.

**What are the use cases for this feature?**
By utilising the display=swap, which is [quite new to google fonts](https://addyosmani.com/blog/google-fonts-font-display/), users will enjoy a [performance boost](https://twitter.com/hecktarzuli/status/1129040263316430850). 

The downside is that [it can cause a split second of FOUC](https://twitter.com/ryantownley/status/1134140895098429440), where the fall back fonts are used before the font downloads, so it would need to be something we could set ourselves from inside the builder, not something added globally via the bloat eliminator settings. Users can choose if it's worth the FOUC for a better speed time, and choose to only use on specific fonts that appear further down the page for eg.

**Comment by wplit** [comment]
Discussion on controlling font-display to typekit via [webfontloader](https://github.com/typekit/webfontloader/issues/409)

**Comment by Nemesis19** [comment]
@wplit Has this issue been taken into consideration?

**Comment by adrien-robert** [comment]
+1

The topic on Facebook I created (initially asking if we could add font-display inside Elegant Custom Fonts plugin by Louis) https://www.facebook.com/groups/1626639680763454/permalink/2399755080118573/

**Comment by tottello** [comment]
Any time soon?

**Comment by stafi** [comment]
Very strange that this has not been corrected. It takes a small amount of time to change.

**Comment by nightgolfer** [comment]
Any news on this? Have a client that religiously tests his site in Lighthouse, and this topic keeps coming up.

**Comment by stafi** [comment]
For now just  use 

plugin called:

Fast Velocity Minify

It helped in my case.

**Comment by nightgolfer** [comment]
Thank you Stafi, I'll give that a try =)


> For now just use
> 
> plugin called:
> 
> Fast Velocity Minify
> 
> It helped in my case.

**Comment by KnickelsTheEkidna** [comment]
Can this please be included in the next release, this can make or break a lot of CLS issues people may be experiencing.

**Comment by israeljernigan** [comment]
Yes, please. Would love to be able to append the new swap attribute for Google Fonts.

**Comment by gerogr** [comment]
+1

**Comment by RoninReilly** [comment]
+1

**Comment by rrdirnens** [comment]
+1

**Comment by falydev** [comment]
`
// Inject dispaly=swap to Google Fonts
function google_fonts_ds_inject_display_swap($html) {

    // Remove existing display swaps
    $html = str_replace("&#038;display=swap", "", $html);
	
	// Add font-display=swap as a querty parameter to Google fonts
    $html = str_replace("googleapis.com/css?family", "googleapis.com/css?display=swap&family", $html);
    $html = str_replace("googleapis.com/css2?family", "googleapis.com/css2?display=swap&family", $html);

    // Fix for Web Font Loader
    $html = preg_replace("/(WebFontConfig\['google'\])(.+[\w])(.+};)/", '$1$2&display=swap$3', $html);

    return $html;
  
}

// Capture HTML
function google_fonts_ds_capture_html() {
    ob_start("google_fonts_ds_inject_display_swap");
}

add_action('init', 'google_fonts_ds_capture_html', 1);
`

**Comment by gugl** [comment]
Any progress there? Can we somehow vote on this issue?
